### PR TITLE
IColunm::hasEqualValues()

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add `hasEqualValues()` interface function to IColumn. It's a building block for other features. cc @KochetovNicolai 

Detailed description / Documentation draft:

.

### CODE REVIEW

**Review by KochetovNicolai — APPROVED** [review]

File: src/Columns/ColumnLowCardinality.cpp [hunk 1]
```diff
@@ -311,6 +311,11 @@ void ColumnLowCardinality::compareColumn(const IColumn & rhs, size_t rhs_row_num
             compare_results, direction, nan_direction_hint);
 }
 
+bool ColumnLowCardinality::hasEqualValues() const
```
Comment (alexey-milovidov) [inline src/Columns/ColumnLowCardinality.cpp:314]: The method can be more efficient, ask @KochetovNicolai

File: src/Columns/ColumnLowCardinality.cpp [hunk 2]
```diff
@@ -311,6 +311,11 @@ void ColumnLowCardinality::compareColumn(const IColumn & rhs, size_t rhs_row_num
...
+{
+    return getDictionary().size() <= 1;
```
Comment (KochetovNicolai) [inline src/Columns/ColumnLowCardinality.cpp:316]: Well, actually, this is not correct.
It is possible that not all rows of dictionary are used.
We still need to check that columns with indexes has same value.

Comment (amosbird) [reply src/Columns/ColumnLowCardinality.cpp:316]: Ok...

**Comment by alexey-milovidov** [comment]
> test_replace_partition/test.py::test_replace_after_replace_failover

There are 5 failures in the last 100 days - flaky test.

**Pull request merged**